### PR TITLE
3.0 plugin load shell

### PR DESCRIPTION
**Pull request by bobmulder**
Implemented the Plugin Shell (https://github.com/cakephp/plugin-installer/issues/28).

The idea was to rebuild the PluginAssets Shell but @ADmad and I decided to do it this way for now.

Example:

```
// loading a plugin
bin/cake plugin load MyPlugin

// loading a plugin including bootstrap and routes
bin/cake plugin load -b -r MyPlugin

// unloading plugin
bin/cake plugin unload MyPlugin
```

I will fork the docs and add this to the documentation.

Tests for the tasks are working.

### CODE REVIEW

File: src/Shell/PluginShell.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * @since         3.0.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ * @author        Bob Mulder <bobmulder@outlook.com>
+ */
+
+namespace Cake\Shell;
+
+use Cake\Console\Shell;
+
+//use Cake\Core\Plugin;
```
Comment (ADmad) [inline src/Shell/PluginShell.php]: These comments needs to be removed.

File: src/Shell/PluginShell.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+//use Cake\Filesystem\Folder;
+//use Cake\Utility\Inflector;
+
+/**
+ * Shell for symlinking / copying plugin assets to app's webroot.
```
Comment (ADmad) [inline src/Shell/PluginShell.php]: This needs to be updated to something like: "Shell for tasks related to plugins."

File: src/Shell/PluginShell.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
+<?php
+
```
Comment (ADmad) [inline src/Shell/PluginShell.php]: Extra empty line

File: src/Shell/PluginShell.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+ * Shell for symlinking / copying plugin assets to app's webroot.
+ *
+ */
+class PluginShell extends Shell
+{
+
+    /**
+     * Tasks to load
+     *
+     * @var type
```
Comment (ADmad) [inline src/Shell/PluginShell.php]: type => array

File: src/Shell/Task/LoadTask.php [hunk 5]
```diff
@@ -0,0 +1,95 @@
+<?php
+
```
Comment (ADmad) [inline src/Shell/Task/LoadTask.php]: Missing docblock

File: src/Shell/Task/LoadTask.php [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+namespace Cake\Shell\Task;
+
+use Cake\Console\Shell;
+use Cake\Core\App;
+use Cake\Filesystem\File;
+
+class LoadTask extends Shell
```
Comment (ADmad) [inline src/Shell/Task/LoadTask.php]: Missing docblock

**Comment by ADmad** [comment]
Please review docblocks for all methods. Many are missing or inaccurate.

File: src/Shell/PluginShell.php (hunk 1 above)
Comment (bobmulder) [inline src/Shell/PluginShell.php]: @ADmad What comments do you mean? 7 till 16? Thought they were required?

Comment (ADmad) [inline src/Shell/PluginShell.php]: The commented code. It's not required :)

**Comment by bobmulder** [comment]
Sorry, my whole branch was a mess. I created a new one and will commit the correct clean code there...

**Pull request closed without merging**